src/main/java/org/synyx/urlaubsverwaltung/application/statistics/ApplicationForLeaveStatisticsService.java

```diff
@@ -40,7 +44,10 @@
     List<ApplicationForLeaveStatistics> getStatistics(FilterPeriod period) {
         final List<VacationType> activeVacationTypes = vacationTypeService.getActiveVacationTypes();
         return getRelevantPersons().stream()
-            .map(person -> applicationForLeaveStatisticsBuilder.build(person, period.getStartDate(), period.getEndDate(), activeVacationTypes))
+            .map(person -> {
+                final PersonBasedata personBasedata = personBasedataService.getBasedataByPersonId(person.getId()).orElse(null);
```

Comment:
can we split this into two mapping so it is clearer what is happening here?

Reply:
Do you mean something like this?
```
final Optional<PersonBasedata> optional = personBasedataService.getBasedataByPersonId(person.getId());
                return optional.ifPresentOrElse(
                    personBasedata -> applicationForLeaveStatisticsBuilder.build(person, personBasedata, period.getStartDate(), period.getEndDate(), activeVacationTypes),
                    () -> applicationForLeaveStatisticsBuilder.build(person, null, period.getStartDate(), period.getEndDate(), activeVacationTypes));
            }
```

or two different build methods (with and without basedata?)

Reply:
Should have taken a deeper look and provide more information. 

We should not pass a null value into a method. If we do that we cannot go with a optimistic programming, like we are doing, because we are never sure of the method variable is null or not. So we need to add a guard and so on. That's now how we should program. So either we decide to overload the method without the `personBasedata` or build a builder pattern or something like that.